Classic_Algorithms/Sorting.py

```diff
@@ -0,0 +1,39 @@
+def merge_sort(lst: list) -> list:
+    if len(lst) > 1:
+        middle_number = len(lst) // 2
+        first_part = lst[:middle_number]
+        second_part = lst[middle_number:]
+
+        merge_sort(first_part)
+        merge_sort(second_part)
+
+        x, y, z = 0, 0, 0
+
+        while x < len(first_part) and y < len(second_part):
```

Comment:
I don't like this 3 'while'. They hugely overlap each other. Can you find another solution?